File: generated_code/python/test_unittest_2299.py
import os
import importlib.util
import unittest

# Dynamic loading of the solution module
solution_path = os.environ.get("TEST_SOLUTION_FILE")
if not solution_path:
    raise ValueError("TEST_SOLUTION_FILE environment variable is not set.")

spec = importlib.util.spec_from_file_location("Solution", solution_path)
sol_module = importlib.util.module_from_spec(spec)
spec.loader.exec_module(sol_module)
Solution = sol_module.Solution
ListNode = sol_module.ListNode


class TestMergeNodes(unittest.TestCase):
    def helper_create_linked_list(self, arr):
        """Helper method to create a linked list from an array."""
        if not arr:
            return None
        head = ListNode(arr[0])
        current = head
        for val in arr[1:]:
            current.next = ListNode(val)
            current = current.next
        return head

    def helper_linked_list_to_list(self, head):
        """Helper method to convert a linked list back to a Python list."""
        result = []
        current = head
        while current:
            result.append(current.val)
            current = current.next
        return result

    def test_example_1(self):
        """Tests Example 1: head = [0, 3, 1, 0, 4, 5, 2, 0] -> [4, 11]"""
        input_list = [0, 3, 1, 0, 4, 5, 2, 0]
        expected_output = [4, 11]

        head = self.helper_create_linked_list(input_list)
        sol = Solution()
        modified_head = sol.mergeNodes(head)
        output_list = self.helper_linked_list_to_list(modified_head)

        self.assertEqual(output_list, expected_output)

    def test_example_2(self):
        """Tests Example 2: head = [0, 1, 0, 3, 0, 2, 2, 0] -> [1, 3, 4]"""
        input_list = [0, 1, 0, 3, 0, 2, 2, 0]
        expected_output = [1, 3, 4]

        head = self.helper_create_linked_list(input_list)
        sol = Solution()
        modified_head = sol.mergeNodes(head)
        output_list = self.helper_linked_list_to_list(modified_head)

        self.assertEqual(output_list, expected_output)

    def test_minimum_constraints(self):
        """Tests the minimum possible valid length constraint (3 nodes: [0, X, 0])."""
        input_list = [0, 5, 0]
        expected_output = [5]

        head = self.helper_create_linked_list(input_list)
        sol = Solution()
        modified_head = sol.mergeNodes(head)
        output_list = self.helper_linked_list_to_list(modified_head)

        self.assertEqual(output_list, expected_output)

    def test_nodes_with_zero_value(self):
        """Tests segments containing non-zero values that are actually 0, but separated (e.g., Node.val == 0 but not consecutive 0's)."""
        input_list = [0, 0, 0]  # The constraints say 'no two consecutive nodes with Node.val == 0' (except bounding zeros, meaning this exact case is invalid per rules, but a segment summing to 0 is possible if nodes are 0? Wait, constraint says '0 <= Node.val <= 1000' and 'no two consecutive nodes with Node.val == 0'. Thus, zeros only exist as delimiters. Let's test single large numbers instead.)
        pass

    def test_single_element_segments(self):
        """Tests multiple segments where each segment contains exactly one node."""
        input_list = [0, 10, 0, 20, 0, 30, 0]
        expected_output = [10, 20, 30]

        head = self.helper_create_linked_list(input_list)
        sol = Solution()
        modified_head = sol.mergeNodes(head)
        output_list = self.helper_linked_list_to_list(modified_head)

        self.assertEqual(output_list, expected_output)

    def test_large_segment_values(self):
        """Tests values at the upper constraint limit (Node.val up to 1000)."""
        input_list = [0, 1000, 1000, 1000, 0, 500, 500, 0]
        expected_output = [3000, 1000]

        head = self.helper_create_linked_list(input_list)
        sol = Solution()
        modified_head = sol.mergeNodes(head)
        output_list = self.helper_linked_list_to_list(modified_head)

        self.assertEqual(output_list, expected_output)


if __name__ == "__main__":
    unittest.main()

File: generated_code/cpp/test_unittest_2299.cpp
#include <gtest/gtest.h>
#include <vector>
#include "solution_proxy.h"

// Helper function to create a linked list from a vector
ListNode* createList(const std::vector<int>& values) {
    if (values.empty()) return nullptr;
    ListNode* head = new ListNode(values[0]);
    ListNode* current = head;
    for (size_t i = 1; i < values.size(); ++i) {
        current->next = new ListNode(values[i]);
        current = current->next;
    }
    return head;
}

// Helper function to convert a linked list to a vector
std::vector<int> listToVector(ListNode* head) {
    std::vector<int> result;
    ListNode* current = head;
    while (current != nullptr) {
        result.push_back(current->val);
        current = current->next;
    }
    return result;
}

// Helper function to free the allocated linked list memory
void freeList(ListNode* head) {
    while (head != nullptr) {
        ListNode* temp = head;
        head = head->next;
        delete temp;
    }
}

class MergeNodesTest : public ::testing::Test {
protected:
    Solution solution;
};

// Test Case: Example 1 from problem description
TEST_F(MergeNodesTest, Example1) {
    std::vector<int> input = {0, 3, 1, 0, 4, 5, 2, 0};
    std::vector<int> expected = {4, 11};
    
    ListNode* head = createList(input);
    ListNode* resultHead = solution.mergeNodes(head);
    std::vector<int> result = listToVector(resultHead);
    
    EXPECT_EQ(result, expected);
    freeList(resultHead);
}

// Test Case: Example 2 from problem description
TEST_F(MergeNodesTest, Example2) {
    std::vector<int> input = {0, 1, 0, 3, 0, 2, 2, 0};
    std::vector<int> expected = {1, 3, 4};
    
    ListNode* head = createList(input);
    ListNode* resultHead = solution.mergeNodes(head);
    std::vector<int> result = listToVector(resultHead);
    
    EXPECT_EQ(result, expected);
    freeList(resultHead);
}

// Test Case: Minimum constraints (3 nodes: [0, X, 0])
TEST_F(MergeNodesTest, MinimumNodes) {
    std::vector<int> input = {0, 5, 0};
    std::vector<int> expected = {5};
    
    ListNode* head = createList(input);
    ListNode* resultHead = solution.mergeNodes(head);
    std::vector<int> result = listToVector(resultHead);
    
    EXPECT_EQ(result, expected);
    freeList(resultHead);
}

// Test Case: Node values are 0 (e.g., [0, 0, 0] is invalid per constraints, but nodes between 0s can be 0)
TEST_F(MergeNodesTest, ZeroValuesBetweenZeros) {
    std::vector<int> input = {0, 0, 0, 0}; // Wait, constraint says "no two consecutive nodes with Node.val == 0"
    // So valid input with 0 value between 0s must be separated like: [0, 0, 1, 0] -> invalid consecutive.
    // "0 <= Node.val <= 1000". If value is 0, it cannot be consecutive. Thus, {0, 0...} is invalid.
    // Correct interpretation of node values being 0 means an actual sequence like [0, 5, 0, 0] is invalid.
    // However, if Node.val is 0, it acts as a separator. The constraint says "no two consecutive nodes with Node.val == 0".
    // This implies all block separators are non-empty. So values between 0s are strictly > 0 due to the constraint.
    // Let's test a single element that reaches max constraint value.
    std::vector<int> inputMaxVal = {0, 1000, 1000, 0, 1000, 0};
    std::vector<int> expectedMaxVal = {2000, 1000};

    ListNode* head = createList(inputMaxVal);
    ListNode* resultHead = solution.mergeNodes(head);
    std::vector<int> result = listToVector(resultHead);
    
    EXPECT_EQ(result, expectedMaxVal);
    freeList(resultHead);
}

// Test Case: Large number of nodes to ensure no stack overflow or timeout (within limit)
TEST_F(MergeNodesTest, LargeList) {
    std::vector<int> input = {0};
    std::vector<int> expected;
    
    // Creating a chain of 10000 blocks of [1, 0] -> 20001 nodes total
    for (int i = 0; i < 10000; ++i) {
        input.push_back(1);
        input.push_back(0);
        expected.push_back(1);
    }
    
    ListNode* head = createList(input);
    ListNode* resultHead = solution.mergeNodes(head);
    std::vector<int> result = listToVector(resultHead);
    
    EXPECT_EQ(result, expected);
    freeList(resultHead);
}

// Test Case: Multiple nodes between zeros summing up to a large value
TEST_F(MergeNodesTest, ManyNodesInSingleBlock) {
    std::vector<int> input = {0};
    int expected_sum = 0;
    for (int i = 0; i < 1000; ++i) {
        input.push_back(5);
        expected_sum += 5;
    }
    input.push_back(0);
    std::vector<int> expected = {expected_sum};
    
    ListNode* head = createList(input);
    ListNode* resultHead = solution.mergeNodes(head);
    std::vector<int> result = listToVector(resultHead);
    
    EXPECT_EQ(result, expected);
    freeList(resultHead);
}